wolfcrypt/test/test.c

```diff
@@ -3003,8 +3035,311 @@ WOLFSSL_TEST_SUBROUTINE int sha512_test(void)
 
     return ret;
 }
+
+#if !defined(WOLFSSL_NOSHA512_224) && \
+     !defined(HAVE_FIPS)
```

Comment:
Same. Why not `(!defined(HAVE_FIPS) || FIPS_VERSION_GE(5, 2)) && !defined(HAVE_SELFTEST)`?

Reply:
It failed to compile in Jenkins with many `unresolved external symbol` linking errors with that logic. Shall I put that back so we can more readily see the errors in Jenkins? Alternatively, I can cleanup this PR commit history for a merge and we can do the conditional FIPS gating as a separate exercise. Either works for me.